tgui/packages/tgui/interfaces/BarberingMenu.js

```diff
@@ -0,0 +1,119 @@
+    <Window
+      width={700}
+      height={500}
+      title="Barber">
+
+      <Window.Content scrollable>
+        <HairOptions />
+        <hr />
+        <Stack>
+          <Stack.Item width="65%">
+            <Stack wrap="wrap" justify="space-around">
+              <Stack width="100%" inline justify="space-around">
+                <Icon name="magnifying-glass" />
+                <Stack.Item mb="10px" grow>
+                  <Input width="100%" onInput={(e, value) => setSearchText(value)} />
+                </Stack.Item>
+              </Stack>
+              <HairPreviewList search_text={searchText}
+                all_hair_names={styles_keys}
+                all_hair_styles={available_styles} />
+            </Stack>
+          </Stack.Item>
+          <Stack.Item position="fixed" width="32%">
+            <PreviewWindow />
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const HairPreviewList = function (props, context) {
+  const { search_text, all_hair_names, all_hair_styles } = props;
+
+  const filtered_list = all_hair_names.filter((x) => x.toLowerCase().includes(search_text.toLowerCase()));
+  return filtered_list.map((value, index) => (<HairPreview
+    key={all_hair_styles[value].id}
+    hair_style={all_hair_styles[value]}
+    hair_name={value} />));
+};
+
+const HairOptions = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { selected_hair_portion } = data;
+
+  return (
+    <Stack justify="space-between" fontSize="15px">
+      <Button.Checkbox checked={selected_hair_portion === "bottom" ? 1 : 0} onClick={() => act("change_hair_portion", { "new_portion": "bottom" })}>Bottom Hair</Button.Checkbox>
+      <Button.Checkbox checked={selected_hair_portion === "middle" ? 1 : 0} onClick={() => act("change_hair_portion", { "new_portion": "middle" })}>Middle Hair</Button.Checkbox>
+      <Button.Checkbox checked={selected_hair_portion === "top" ? 1 : 0} onClick={() => act("change_hair_portion", { "new_portion": "top" })}>Top Hair</Button.Checkbox>
+      <hr />
+      <Button bold color="red" icon="cut" onClick={() => act("do_hair", { "style_id": null })}>Create Wig</Button>
+    </Stack>
+  );
+};
+
+const HairPreview = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { hair_style, hair_name } = props;
+  return (
+    <Section width="140px" direction="column" align="center">
+      <Image pixelated width="60px" height="100px" src={`${hair_style["hair_icon"]}`} />
+      <Box width="100%" fontSize="15px" textAlign="center" mb="10px">{hair_name}</Box>
+      <Stack inline justify="space-between">
+        <Button color="blue" height="20px" icon="cut" onClick={() => act("do_hair", { "style_id": hair_style["hair_id"] })}>Cut</Button>
+        <Button color="blue" height="20px" icon="eye" onClick={() => act("update_preview", { "action": "new_hair", "style_id": hair_style["hair_id"] })}>Preview</Button>
+      </Stack>
```

Comment:
A `Stack` should probably have `Stack.Item`s as children, rather than random other things (`Button`s, in this case). While I'm sure it works, I believe it's unintended behavior. Not 'defined' behavior, at least.

Reply:
I attempted to cover most of the elements under `<Stack.Item>` where it should be. Tell me if it's good or nah, or if I was excessive when turning everything into `<Stack.Item>`